src/main/java/com/couchbase/lite/BlobStore.java

```diff
@@ -41,28 +45,36 @@
  * @exclude
  */
 public class BlobStore {
-
-    public static String FILE_EXTENSION = ".blob";
-    public static String TMP_FILE_EXTENSION = ".blobtmp";
-    public static String TMP_FILE_PREFIX = "tmp";
+    public static final String FILE_EXTENSION = ".blob";
+    public static final String TMP_FILE_EXTENSION = ".blobtmp";
+    public static final String TMP_FILE_PREFIX = "tmp";
+    public static final String ENCRYPTION_ALGORITHM = "AES";
+    public static final String ENCRYPTION_MARKER_FILENAME = "_encryption";
 
     private String path;
+    private SymmetricKey encryptionKey;
 
-    public BlobStore(String path) {
-        this(path, false);
+    public BlobStore(String path, SymmetricKey encryptionKey) throws CouchbaseLiteException {
+        this(path, encryptionKey, false);
     }
 
-    public BlobStore(String path, boolean autoMigrate) {
+    public BlobStore(String path, SymmetricKey encryptionKey, boolean autoMigrate)
+            throws CouchbaseLiteException {
         this.path = path;
-        File directory = new File(path);
+        this.encryptionKey = encryptionKey;
 
-        if (!directory.exists()) {
-            if (!directory.mkdirs())
-                Log.w(Log.TAG_DATABASE, "Unable to make directory: %s", directory);
-        }
-        if (!directory.isDirectory()) {
-            throw new IllegalStateException(String.format("Unable to create directory for: %s",
-                    directory));
+        File directory = new File(path);
+        if (directory.exists()) {
+            if (!directory.isDirectory())
+                throw new CouchbaseLiteException("BlobStore: Blobstore is not a directory", Status.ATTACHMENT_ERROR);
+            verifyExistingStore();
```

Comment:
In general cases, there shouldn't be the case. Jens added this to fix a bug that the iOS doesn't really encrypting the attachments.

VerifyingExistingStore does 2 things for ensuring things work as expected and if not (for some reasons such as bugs), instead of failing silently, an error will be thrown:
1. Verifying the marker file exists otherwise error should be thrown.
2. Verifying an encryption algorithm.